airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py

```diff
@@ -280,6 +277,149 @@ def test_secret_is_properly_marked(self, connector_spec_dict: dict, detailed_log
+            for path, _value in dpath.util.search(specification, f"**/properties/{forbidden_key}", yielded=True):
+                found_keys.remove(path)
+
+        if len(found_keys) > 0:
+            key_list = ", ".join(found_keys)
+            pytest.fail(f"Found the following disallowed JSON schema features: {key_list}")
+
+    def test_date_pattern(self, connector_spec_dict: dict, detailed_logger):
+        """
+        Properties with format date or date-time should always have a pattern defined how the date/date-time should be formatted
+        that corresponds with the format the datepicker component is creating.
+        """
+        specification = connector_spec_dict["connectionSpecification"]
+        for format_path, format in dpath.util.search(specification, "**/format", yielded=True):
+            if not isinstance(format, str):
+                # format is not a format definition here but a property named format
+                continue
+            if format == "date" or format == "date-time":
+                property_definition = self._get_parent(specification, format_path)
+                pattern = property_definition.get("pattern")
+                if format == "date" and not pattern == DATE_PATTERN:
+                    detailed_logger.warning(
+                        f"{format_path} is defining a date format without the corresponding pattern. Consider setting the pattern to {DATE_PATTERN} to make it easier for users to edit this field in the UI."
+                    )
+                if format == "date-time" and not pattern == DATETIME_PATTERN:
+                    detailed_logger.warning(
+                        f"{format_path} is defining a date-time format without the corresponding pattern Consider setting the pattern to {DATETIME_PATTERN} to make it easier for users to edit this field in the UI."
+                    )
+
+    def test_date_format(self, connector_spec_dict: dict, detailed_logger):
+        """
+        Properties with a pattern that looks like a date should have their format set to date or date-time.
+        """
+        specification = connector_spec_dict["connectionSpecification"]
+        for pattern_path, pattern in dpath.util.search(specification, "**/pattern", yielded=True):
+            if not isinstance(pattern, str):
+                # pattern is not a pattern definition here but a property named pattern
+                continue
+            if pattern == DATE_PATTERN or pattern == DATETIME_PATTERN:
+                property_definition = self._get_parent(specification, pattern_path)
+                format = property_definition.get("format")
+                if not format == "date" and pattern == DATE_PATTERN:
+                    detailed_logger.warning(
+                        f"{pattern_path} is defining a pattern that looks like a date without setting the format to `date`. Consider specifying the format to make it easier for users to edit this field in the UI."
```

Comment:
should this fail? What would be a valid format that would trigger a warning?

Reply:
I was on the fence about this, too - I decided this way because of the following reasoning:
* Showing a date picker for a field instead of requiring the user to type it out is more of a presentational/convenience thing than a "just plain wrong/not configurable at all in the UI" thing (like the other checks)
* Semantically, a pattern doesn't necessarily mean it's a date - it is very likely for this specific pattern (hence the warning), but is it the right call the enforce this relationship?
* One example could be a sort of serial number which has the same format, but is semantically not a date - you wouldn't want to force a connector developer to render a date picker for that.

This is definitely not a strong reasoning, I will put it in the issue I mentioned above about the check the other way.

Reply:
+1 to @flash1293's reasoning - I had similar reasoning [here](https://github.com/airbytehq/airbyte/issues/18741) re: we can't enforce date-formatting on things that just look like they could be dates - its up to the source to define whether we should interpret it as a date or not